internal/command/init.go

```diff
@@ -1379,3 +1423,25 @@ The current .terraform.lock.hcl file only includes checksums for %s, so OpenTofu
 To calculate additional checksums for another platform, run:
   tofu providers lock -platform=linux_amd64
 (where linux_amd64 is the platform to generate)`
+
+const implicitProviderReferenceHead = `Automatically-inferred provider dependency
+	on %s line %d:
+		%d: %s "%s" "%s"`
+
+const implicitProviderReferenceBody = `Due to the prefix of the resource type name OpenTofu guessed that
+you intended to associate %s with a provider whose
+local name is "%s", but that name is not declared in this module's
+required_providers block. OpenTofu therefore guessed that you
+intended to use %s, but that provider does not exist.
+
+Make at least one of the following changes to tell OpenTofu which
+provider to use:
+
+- Add a declaration for local name "%s" to this module's
+required_providers block, specifying the full source address
+for the provider you intended to use.
+- Verify that "%s" is the correct resource type name to use.
+Did you omit a prefix which would imply the correct provider?
+- Use a "provider" argument within this resource block to override
+OpenTofu's automatic selection of the local name "%s".
+`
```

Comment:
The diagnostic renderer does its own line-wrapping, so in order for this to be displayed correctly at all screen sizes you'd need to remove all of the "soft-wrapping" -- that is, all of the newlines that aren't separating paragraphs or list items.

That's particularly important in this case where we're substituting user-provided symbols into the message and so we can't actually predict how long the lines will be as part of this constant. This problem is already visible in one of the test cases above, for example:

https://github.com/opentofu/opentofu/blob/c12c63eaf5a81fd4fa95089c3e9d5cf79ffef07d/internal/command/e2etest/init_test.go#L519-L545

Notice that "module's" got wrapped onto a line of its own because "nonexistingprov" made that line too long.

Reply:
Thanks a lot. Applied 6237d518c20e419f4321dad86f1d741a1724a335